compiler/src/dotty/tools/dotc/transform/HoistSuperArgs.scala

```diff
@@ -140,7 +140,7 @@ class HoistSuperArgs extends MiniPhase with IdentityDenotTransformer { thisPhase
                 }
               },
               treeMap = {
-                case tree: RefTree if paramSyms.contains(tree.symbol) =>
+                case tree: RefTree if tree.symbol.isParamOrAccessor =>
```

Comment:
It's better to strengthen the condition a little:

```Scala
-                case tree: RefTree if paramSyms.contains(tree.symbol) =>
+                case tree: RefTree if needsRewire(tree.tpe.asInstanceOf[NamedType]) =>
```
where `needsRewire` is the condition in the `typeMap`:

```Scala
            def needsRewire(tp: NamedType) =
                (tp.symbol.owner == cls || tp.symbol.owner == constr) && tp.symbol.isParamOrAccessor
```

Reply:
Totally agree, although all tests pass it is better to make it more strict. Thanks for the suggestion.

One question, why should we cast this to NamedType instead of pattern match like: 
```
def needsRewire(t: Type): Boolean = t match {
        case tp: NamedType => (tp.symbol.owner == cls || tp.symbol.owner == constr) && tp.symbol.isParamOrAccessor
        case _             => false
      }
```
Can the RefTree type (Select, Ident) never be anything else than a TermRef or TypeRef? if that's the case then we can 'safely' cast this. And safe a pattern match :)

Reply:
Yes, it's better to do a pattern match (the cast should be safe, though). And we can remove duplicate code in `typeMap` to reuse `needsRewire`.

Reply:
Thanks, I have applied your suggestion, and removed the duplicated code in the typeMap.